src/write_gcm.c

```diff
@@ -2364,54 +2347,60 @@ static monitored_resource_t *parse_monitored_resource(char *metadata,
       .yajl_double = NULL,
       .yajl_number = NULL,
       .yajl_string = &wg_mr_handle_string,
-      .yajl_start_map = &wg_mr_handle_start_map,
+      .yajl_start_map = NULL,
       .yajl_map_key = &wg_mr_handle_map_key,
       .yajl_end_map = &wg_mr_handle_end_map,
       .yajl_start_array = NULL,
       .yajl_end_array = NULL
     };
 
-    wg_mr_parse_context_t parse_context = {
-      .processing_labels = 0,
-      .total_labels = 0
+    wg_mr_parse_context_t ctx = {
+      .processing_labels = 0
     };
 
     #if defined(YAJL_MAJOR) && YAJL_MAJOR >= 2
-      handle = yajl_alloc(&callbacks, NULL, &parse_context);
+      handle = yajl_alloc(&callbacks, NULL, &ctx);
     #else
       yajl_parser_config config = { 0, 1 };
-      handle = yajl_alloc(&callbacks, &config, NULL, &parse_context);
+      handle = yajl_alloc(&callbacks, &config, NULL, &ctx);
     #endif
 
-    if(yajl_parse(handle, (const unsigned char*) metadata, strlen(metadata)) != yajl_status_ok) {
+    int stat = yajl_parse(handle, (const unsigned char *) metadata, 
+                          strlen(metadata));
+
+    if (stat != yajl_status_ok) {
       ERROR("write_gcm: parse_monitored_resource: Invalid JSON response.");
       goto finish;
     }
 
+    // The definition for parsing has encountered a breaking change between
+    // yajl v1 and yajl v2 despite being fundamentally equal in its behavior.
     #if defined(YAJL_MAJOR) && YAJL_MAJOR >= 2
       int parse_result = yajl_complete_parse(handle);
     #else
       int parse_result = yajl_parse_complete(handle);
     #endif
 
-    if(parse_result != yajl_status_ok) {
+    if (parse_result != yajl_status_ok) {
       ERROR("write_gcm: parse_monitored_resource: Invalid JSON response.");
       goto finish;
     }
 
-    char *type = parse_context.type;
-    label_t *labels = calloc(parse_context.total_labels, sizeof(label_t));
+    char *type = ctx.type;
+    int total_labels = llist_size(ctx.label_list);
+    label_t labels[total_labels];
 
-    wg_mr_label_node_t *curr = parse_context.head;
-
-    for(int i = parse_context.total_labels - 1; i >= 0; i--) {
-      labels[i].key = curr->label.key;
-      labels[i].value = curr->label.value;
-      curr = curr->next;
+    llentry_t *curr = llist_head(ctx.label_list);
+    for(int i = 0; curr != NULL; curr = curr->next, ++i) {
+      labels[i].key = curr->key;
+      labels[i].value = (char *) curr->value;
     }
 
     response = monitored_resource_create_from_array(type, project_id, labels,
-                                                    parse_context.total_labels);
+                                                    total_labels);
+
+    llist_destroy(ctx.label_list);
```

Comment:
You're leaking the keys and label values, aren't you? `llist_destroy` will only free the entries, not their content.